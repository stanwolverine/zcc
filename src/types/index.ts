import type { USER_TYPES } from '@/constants/UserTypes';

export type TableZellerCustomerFilterInput = Partial<{
  id: TableStringFilterInput | null;
  name: TableStringFilterInput | null;
  email: TableStringFilterInput | null;
  role: TableStringFilterInput | null;
}>;

export type TableStringFilterInput = Partial<{
  ne: string | null;
  eq: string | null;
  le: string | null;
  lt: string | null;
  ge: string | null;
  gt: string | null;
  contains: string | null;
  notContains: string | null;
  between: [string] | null;
  beginsWith: string | null;
}>;

export interface Variables {
  filter: TableZellerCustomerFilterInput | null;
  limit: number | null;
  nextToken: string | null;
}

export type ZellerCustomer = {
  id: string;
  name: string | null;
  email: string | null;
  role: string | null;
};

export type ZellerCustomerConnection = {
  items: ZellerCustomer[];
  nextToken: string;
};

export type ZellerCustomerData = {
  listZellerCustomers: ZellerCustomerConnection;
};

export type UserType = (typeof USER_TYPES)[number];
