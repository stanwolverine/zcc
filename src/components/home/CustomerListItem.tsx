import React from 'react';
import { StyleSheet, useColorScheme } from 'react-native';

import { ThemedText } from '@/components/ThemedText';
import { ThemedView } from '@/components/ThemedView';
import { ZellerCustomer } from '@/types';
import { Colors } from '@/constants/Colors';

export type CustomerListItemProps = {
  data: ZellerCustomer;
};

const _CustomerListItem: React.FC<CustomerListItemProps> = ({ data }) => {
  const colorScheme = useColorScheme() ?? 'light';

  return (
    <ThemedView testID={`customer-item-${data.id}`} style={styles.container}>
      {/** Avatar */}
      <ThemedView
        testID={`customer-avatar-${data.id}`}
        style={[
          styles.avatar,
          {
            backgroundColor: Colors[colorScheme].secondary,
          },
        ]}>
        <ThemedText
          type="small"
          style={[styles.avatarText, { color: Colors[colorScheme].onSecondary }]}>
          {data.name?.charAt(0)}
        </ThemedText>
      </ThemedView>

      {/** Info */}
      <ThemedView>
        <ThemedText testID={`customer-name-${data.name}`} type="default">
          {data.name}
        </ThemedText>

        <ThemedText
          testID={`customer-role-${data.role}`}
          type="small"
          lightColor="gray"
          darkColor="gray"
          style={styles.role}>
          {data.role}
        </ThemedText>
      </ThemedView>
    </ThemedView>
  );
};

export const CustomerListItem = React.memo(_CustomerListItem);

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    gap: 12,
  },
  avatar: {
    minWidth: 45,
    borderRadius: 4,
    aspectRatio: 1.0,
    alignItems: 'center',
    justifyContent: 'center',
  },
  avatarText: { textTransform: 'uppercase' },
  role: { textTransform: 'capitalize' },
});
