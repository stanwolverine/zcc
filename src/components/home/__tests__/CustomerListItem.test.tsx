import { render } from '@testing-library/react-native';

import { CustomerListItem } from '../CustomerListItem';
import { ZellerCustomer } from '@/types';
import { useColorScheme } from 'react-native';

describe('CustomerListItem', () => {
  it('renders correctly', () => {
    // Arrange
    const customer: ZellerCustomer = {
      id: '1',
      email: 'test@test.com',
      name: 'Test',
      role: 'ADMIN',
    };
    jest.mocked(useColorScheme).mockReturnValueOnce('light');

    // Act
    const tree = render(<CustomerListItem data={customer} />).toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
  });
});
