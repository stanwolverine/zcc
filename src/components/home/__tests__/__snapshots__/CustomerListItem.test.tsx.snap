// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerListItem renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "flexDirection": "row",
        "gap": 12,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        [
          {
            "alignItems": "center",
            "aspectRatio": 1,
            "borderRadius": 4,
            "justifyContent": "center",
            "minWidth": 45,
          },
          {
            "backgroundColor": "#E7F2FB",
          },
        ],
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          {
            "fontSize": 14,
            "lineHeight": 21,
          },
          [
            {
              "textTransform": "uppercase",
            },
            {
              "color": "#0171CE",
            },
          ],
        ]
      }
    >
      T
    </Text>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      Test
    </Text>
    <Text
      style={
        [
          {
            "color": "gray",
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          {
            "fontSize": 14,
            "lineHeight": 21,
          },
          {
            "textTransform": "capitalize",
          },
        ]
      }
    >
      ADMIN
    </Text>
  </View>
</View>
`;
